client/blocks/product-selector/index.jsx

```diff
@@ -197,9 +214,10 @@ export class ProductSelector extends Component {
 
 		return map( products, product => {
 			const selectedProductSlug = this.state[ this.getStateKey( product.id, intervalType ) ];
-			const productObject = storeProducts[ selectedProductSlug ];
 			const stateKey = this.getStateKey( product.id, intervalType );
 			const purchase = this.getPurchaseByProduct( product );
+			const upgrade =
+				product.upgradeProductMap && purchase && this.getPurchaseUpgrade( purchase, product );
```

Comment:
How about moving this rather complex condition to a new method like `renderPurchaseUpgrade()`?

In such case the `renderProducts()` method wouldn't have to care if there are any upgrades or not.